### ISSUE TITLE
.trigger("chosen:updated"); does not work with recent versions of jquery.

### ISSUE THREAD

**Issue by brigadier** [harvesthq/chosen#2198]
For example, I use the following code on the demo page http://harvesthq.github.io/chosen/ but with https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js instead of 1.6.4

```
<div>
  <script>
  function ins() {
    var ur = $('option[value="Uruguay"]');
    $(ur).attr("selected", true);
    $(".chosen-select").trigger("chosen:updated");
  };
  </script>
  <button id="ins" onclick="ins(); return false;">Insert</button>
</div>
```
If I click on the button first time, Uruguay gets selected. Then I delete Uruguay by clicking an 'x' at the label. Then I click on the button again, but this time it does not work, the option does not apper any more an the list of selected options. With jquery 1.6.4 (the version used in the demo page) it works fine.

The standalone example: https://github.com/brigadier/chosentest

**Comment by vksvicky** [comment]
I am having the same issue and I am using jQuery v1.11.1

**Comment by jeremybenaim** [comment]
Works on latest Chrome. Which browser are you using?

**Comment by vksvicky** [comment]
ok, I had a bit of code change and now it works on Firefox and Chrome.

My on change event was defined as

$('.chosen-select').on('change', function(event){
//What I should do 
});

And
setInterval(function() {
 $('.chosen-select").trigger("change"); //was ("chosen:updated");
}, 10000); //every 10 seconds
});

**Comment by jeremybenaim** [comment]
This should work too: 

```javascript
$('.chosen-select').on('chosen:updated', function(event){
   // your stuff here
});

window.setInterval(function() {
  $('.chosen-select').trigger('chosen:updated');
}, 1000);
```

**Comment by brigadier** [comment]
I fixed the problem with 

```
vals = ...; //read old selected options values, add new ones
$(select).val(vals);
$(select).trigger("chosen:updated");

```

It works in all browsers.

**Comment by tjschuck** [comment]
Going to close this issue then -- seems like it's not a bug.

**Comment by arce701** [comment]
$(".chosen-select").chosen().trigger("chosen:updated");

**Comment by wuchao** [comment]
Yesterday, $(select).trigger("chosen:updated"); is OK.But today it also doesn't work. The reason is the chosen had updated last night. So if you want to use the above code,you should make sure your chosen version is less than or equals "1.7.0".

**Comment by saadwaseem** [comment]
> I fixed the problem with
> 
> ```
> vals = ...; //read old selected options values, add new ones
> $(select).val(vals);
> $(select).trigger("chosen:updated");
> ```
> 
> It works in all browsers.

This solution is more reliable and does not require rebuilding the chosen again as proposed in one stackoverflow's thread. Thanks.

**Comment by hemanthkumarg70** [comment]
Use below code for multiple selection

$("#single_example").chosen().val(['2', '1']).trigger("chosen:updated");
